tests/test_nblist.py

```diff
@@ -59,59 +62,91 @@ def get_water_coords(D, sort=False):
     return x
 
 
-def test_neighborlist():
+def build_reference_ixn_list(coords: NDArray, box: NDArray, cutoff: float) -> List[List[float]]:
+    # compute the sparsity of the tile
+    ref_ixn_list = []
+    N = coords.shape[0]
 
-    water_coords = get_water_coords(3, sort=False)
-    for size in [35, 64, 129, 1025, 1259, 2029]:
+    block_size = 32
 
-        for nblist in (custom_ops.Neighborlist_f32(size), custom_ops.Neighborlist_f64(size)):
+    num_blocks = (N + block_size - 1) // block_size
+    col_coords = np.expand_dims(coords, axis=0)
 
-            for _ in range(2):
+    box_diag = np.diag(box)
+    for rbidx in range(num_blocks):
+        row_start = rbidx * block_size
+        row_end = min((rbidx + 1) * block_size, N)
+        row_coords = coords[row_start:row_end]
+        row_coords = np.expand_dims(row_coords, axis=1)
 
-                print("testing size:", size)
+        deltas = row_coords - col_coords
+        deltas -= box_diag * np.floor(deltas / box_diag + 0.5)
```

Comment:
nit: could this be a utility function? Maybe something like

```python
def nearest_image(x: NDArray, box_diag: NDArray) -> NDArray:
    return x - box_diag * np.floor(x / box_diag + 0.5)
```

Reply:
Resolved in https://github.com/proteneer/timemachine/pull/726/commits/c42141260bc069e76d470677efef14c91524bb3a